catacombs-chimeras-persistence/src/main/java/cz/muni/fi/dao/RoleDaoImpl.java

```diff
@@ -0,0 +1,51 @@
+package cz.muni.fi.dao;
+
+import static org.apache.commons.lang3.Validate.notNull;
+
+import cz.muni.fi.entity.Role;
+import org.springframework.stereotype.Repository;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.util.List;
+
+/**
+ * @author Ondřej Benkovský
+ */
+@Repository
+public class RoleDaoImpl implements RoleDao{
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    public void create(final Role role) {
+        notNull(role, "role should not be null");
+        entityManager.persist(role);
+    }
+
+    public void delete(final Role role) {
+        notNull(role, "role should not be null");
+        entityManager.remove(role);
+    }
+
+    public Role findById(final Long id) {
+        notNull(id, "id should not be null");
+        return entityManager.find(Role.class, id);
+    }
+
+    public Role findByName(final String name) {
```

Comment:
Umí java nějak naznačit, že ta metoda může vrátit `Role|null` (třeba přes Docblook?)
Něco ve stylu `Role?` nebo tak něco?
Pokud ano, tak imo stačí to, jinak bych dal vyjímku.